### ISSUE TITLE
New Settings Panel

### ISSUE THREAD

**Issue by looterz** [looterz/atmos#10]
This is the biggest task to get Atmos 2 to a state where it can be released and replace Atmos 1.

The idea I had for the new settings panel involved a better user experience for defining sky color patterns over time (and the fog over time), this would be a keyframe animation type editor where you can insert tables of color and fog values at different points in time for a given "phase", which atmos then uses to lerp between during the Time of Day. 

This is how it's always worked, it's just been manually defined instead. Having a keyframe editor type panel where we can create and customize the data within SettingsClass.SkyColors and associate SettingsClass.FogColors with it would be the ideal. Just being able to have a timeline, insert a new "phase" that has all the color and fog data available to edit with color wheels, and scrub across the timeline to visualize it in real-time would be a huge improvement to having it all statically defined as it has been for the last 10 years.

Other than the big ticket item above, just improving the settings area in general vs Atmos 1 would be nice, and making sure almost every setting has an associated cvar for use. Another wishlist item I had for this was that weather's could have their own settings that could be customized by admins as well, and an easier way to configure how weather works in general and blacklist certain weathers etc.

**Comment by ghost** [comment]
yeah im gonna write a per-map config system, if it works out well it should let users override anything on a per-map basis (blacklist, skycolors, fogcolors)

i can try to write a skycolor simulator but that might take some work, i'll see
